systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/AbstractKafkaClient.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.systemtest.kafkaclients;
+
+import io.fabric8.kubernetes.api.model.LoadBalancerIngress;
+import io.fabric8.kubernetes.api.model.Service;
+import io.fabric8.openshift.api.model.Route;
+import io.fabric8.openshift.client.OpenShiftClient;
+
+import java.security.InvalidParameterException;
+
+import static io.strimzi.api.kafka.model.KafkaResources.externalBootstrapServiceName;
+import static io.strimzi.test.k8s.KubeClusterResource.kubeClient;
+
+public abstract class AbstractKafkaClient {
+
+    protected String topicName;
+    protected String namespaceName;
+    protected String clusterName;
+    protected int messageCount;
+    protected String consumerGroup;
+    protected String kafkaUsername;
+    protected String securityProtocol;
+    protected String caCertName;
+    protected KafkaClientProperties clientProperties;
```

Comment:
You've got a builder, so why can't these be `final`?

Reply:
yeah, that's a problem that come in my mind. If you take a close look on how I am using the client for instance:
```
InternalKafkaClient internalKafkaClient = new InternalKafkaClient.Builder()
            .withUsingPodName(kafkaClientsPodName)
            .withTopicName("my-topic-test-3")
            .withNamespaceName(NAMESPACE)
            .withClusterName(kafkaClusterSourceName)
            .withKafkaUsername(userSource.getMetadata().getName())
            .withMessageCount(messagesCount)
            .withConsumerGroupName(CONSUMER_GROUP_NAME + "-" + rng.nextInt(Integer.MAX_VALUE))
            .build();
```

and then after a few lines, I need to set some of the attributes to the new value

```
internalKafkaClient.setClusterName(kafkaClusterTargetName); internalKafkaClient.setKafkaUsername(userTarget.getMetadata().getName());
```
and that's the problem why these properties some of them can't be final. Sure I can rewrite some of them to the final, which is not changed by the setter but that I consider as a not proper way :)

Reply:
It seems strange to me that you want the client first to point to one cluster and then another. It would be more natural if a client, once created, was immutable and you use two clients instance. What you need is to be able to base a builder's state on a pre-existing client, and then tweak it with a few more `with()` calls, so you can build the 2nd client conveniently.

Reply:
In the next PR, I want to create some `default clients`, which will satisfy it. Also, I agree, with immutalibity and will do it.

Reply:
I have already created a task for that.